pkg/templatelookup/regular.go

```diff
@@ -203,7 +205,8 @@ type Lookup interface {
 	WithContext(ctx context.Context) (*ModuleTemplateTO, error)
 }
 
-func NewTemplateLookup(client client.Reader,
+// New returns a new instance of TemplateLookup.
+func New(client client.Reader,
```

Comment:
If you renamed to `New`, it's too generic, and this function only used for regular template, but not mandatory. Since you put both under this  `templatelookup` package, there is no idea for the user calling `templatelookup.New`, is it for regular template, or mandatory, or both?